### ISSUE TITLE
Version 4 documentation

### ISSUE THREAD

**Issue by Gristof** [amzn/style-dictionary#1279]
The official documentation ([website](https://amzn.github.io/style-dictionary/#/config) and [readme](https://github.com/amzn/style-dictionary?tab=readme-ov-file#usage)) is not up to date with the latest v4 release. The official [official release notes](https://github.com/amzn/style-dictionary/releases/tag/v4.0.0) are only partially helpful, when it comes to migrating from v3.

Specifically, I am struggling to get this configuration to work with v4, even though I was working as close as possible to the release notes:

```js
import StyleDictionary from "style-dictionary";

const sd = new StyleDictionary({
  source: ["./tokens/**/*.json"],
  platforms: {
    ts: {
      transformGroup: "js",
      buildPath: "dist/js/",
      files: [
        {
          format: "javascript/es6",
          destination: "variables.js",
        },
        {
          format: "typescript/es6-declarations",
          destination: "variables.d.ts",
        },
      ],
    },
    scss: {
      transformGroup: "scss",
      buildPath: "dist/scss/",
      files: [
        {
          format: "scss/variables",
          destination: "_variables.scss",
        },
      ],
    },
  },
});

sd.registerFormat({
  name: "scss/variables",
  format: function ({ dictionary }) {
    return dictionary.allTokens
      .map((token) => `$${token.name}: ${token.value};`)
      .join("\n");
  },
});

await sd.buildAllPlatforms();
```

1. Am I missing something obvious here?
2. Is it possible to update the official documentation?

**Comment by Gristof** [comment]
Thanks for the quick reply. 

Unfortunetaly there is no error in the CLI. The files seem to be generated. In my `dist` folder, only the `js` files are generated -- but empty, and the `scss` are not at all.

**Comment by jorenbroekema** [comment]
Huh, odd, [I tried your config in the configurator here](https://configurator.tokens.studio/#project=5Vtdb9s2FP0rgrth7RDbEiVRUvYyDEW3vQ3tQzHMxcCPS1utLBminDoN/N9HSbYjyZb1kRDYFuchCSmdQ17ew0vSvA8TmW15mMw+yySe3E4eFrFhLCY8XEMswyReTG6NoiwvlYxEUCnJy+5ItC3KFhO0mNw8VmT3m0O5DL+F8XIxKev2N0e4nWzHcja7gWhy3Y72cOrPbCf3xo9GpaDo1H4g2Zr3IpPr5yCLlr3I1vw5yHZRL7Jo+QSy/FfJqKo2hBV1FT+75hj1oWzjO4KOdBI1bkORe3qEGqShyD2HX43IUOSeY72LhltjG2Xh9ATXSyjDGCu1HCRLw012mK4Wkz+TrcFIbKzIHRhFUzYRGAWzNMLYIIZUGKrogGZkyReIZ8Z7INxYJykYSWxkK5Bwa6yybCNv53OeMDkrnpOzcsqckzsSRoRGMC3L5we4w7+vKkaY1qsuKoEmKYf0PeHhVtan3StO67TYqQ42xKX8UYjXXMnCoyD7+VAVo3SjWkm7Kurkz+JOaQH2XN5UNnBagh59qmSaTo/NKO1z2aFYEiVp3ZVoRNiXdmu+MotPi8UKvPNx+roKsysj9EoUn2GYy5TcNyCtvF2VkjMaTxDBKjRXiB6p8odQBzJwgYQ3BtnuQkbgC3MMstOBzCh3YRSy24FMTAJsFDLuQPYs3wzwGGSvA9khrov9Mch+BzLitueMQg46kC2CTNTfn5sKSoEPE9BDOV3M1Isz5QP7LuZa7Xewy8rYLRscRQArtxjH2NeoL+b6hIfi/kJVnToKlyvFUyO/0BVz5lx8JA/Ch0dWMno0Xs2EZ//sR8wZ/ytj2rqNab8gYyLdxnRekDEt3cbsCobCxepHQzD8LwwTJ+mXf8UoeS/HltqnD//l2FJ7XAteji31LLiai9okJfEShm4MhdoYjtqwdG0MBQBloxb/XRtDQbntcx0bQ4EJd0dFLLdzm+wHNtYQC19xjikaNYJdG0NmuhhZOjaGAXPQOOSujaFHbceC0RvDe4ii5OtwDQldGhKMCi0aUqr3Az0aAtMFLRpiLHCoFg3hQJ026dAQ9bzAElo05LnEwlo05DjIMkdraJkCxEMlZCoJOTokxLDA3NUhoYAApo4OCWGf24GlQ0KOT6nn65CQ7StvDHRICAnfdYmW80kPe06gQ0IIuY7NR0soAxINVBDgfCmnQ0EUCReIDgX5FmCuJQg5gltMy0LO9ikiTIuCrMALXC0KYh7xqBYF+S5gLQs5ZDvgotEKoudfjHYqiApfCC0KArCFlq1QYDIu9MQgm9rAtcQgFARg6VGQj5iWZRyimFFTi4KYi3ykRUHEsQccvzYVFMY8XCbDNeTo0RCzORGgQ0PEpiMjZ5eGPBEwoSUKYezBuJjcpSGXYI/7OjTkMNeiWrZCjtrzB6YODdnMxpiO1tBmm26iwUdyRLh6NAQB9wXXoSF1BsX0aIh6gTMuwnVpKBAe0aMh33QJ17KSUxtOzLQcybmuzQKiQ0OOY9seGa+hMP4yVEFCKcjToSAB3AOkQ0GCqiUG6FCQwL46FdVzqI0doudQ21ULZqZDQdS3ka9lJRd4CGOuQ0GeiRw3GKqg/NfxtmOS37DNqt9oLSbnB93Va6nm9y2XEk9QTaFeu+TtDodbhctVO2DQG7BqBpHE2TuyDqMQ6lc/V0B4/aJ9g+/3OIO0hbGO2uwHTfh9O+77hCZZMgS42p8ojOE3yC+EDOyO1T6+NcxhnbGcQai1nkCmDPzhQr4DB0G2UdbO2nYjtwHZ7EoYsxSIhCtea7W7bQc4h07wqTsMu2qtDUnJMiWb1SWDrax2UrstDekc8UyPqB0W4aGwTVV+bPfi97DcRiS9oqHjA60i+tjmzweKX5LoykiVtcPBc7HobHyO/6wtbw7Kh/Bbc5684lsPp1dmK5QnQFkz5F5hP6AP8bMqhf0ECrsfhfMECqcfhfsECrcfBX4CBe5FkTtiQTKcoSOg4OGIV7PYzttszvwRhrmahHeJBPcnqaoQdmSdJ7LIFeG15VqFsdKK3aWIeF8W1pOh1Pl4elzp1SrkRgUufgnnsMzMK9IlJa/Nm+Jn5rxp6VoYxyqMHVretGA112hXfUj1fz+5maSwDGUeBt++28YszzeSs89ycjsJ15skzYwP2X0Eb8OihqT3hkiTtfGDzEun/FT8w0+L+PDCg3GENPaHp38ur4BND0lGkk+zlMRSJOla5m8u4vncqJXeGCjmRh7V1pAjqUIjKbKhZJEFRcEgnKuwn7/5AcDIc5keM5uWYbba0hlL1vMr1Iv42NLXjV6+UY1StmFJLMJlIytZJtu0vE/210kKj9nLpW0/HdcPKWzShIGUhxyo0yu1ZjVeikhWtq/qhllTCad+/Jom2005xJ/rSWR0G0b8D5Ktylqu+jr/LOe1Z0QYQbVp+ad2GS+XTrom2YGB3JEyM20OEjfu2xVpa1kYk8e0tTuShnk+We5U1at3+5tedLn3PtJN1XJPxfACXg7h5rOszn7889PZrMZkH0OXj3WYOn/oScYuAE696O7w3489Lht4pcenKWD/Dw==) and this does generate the files, so not sure what's going wrong on your end

**Comment by Gristof** [comment]
Seems like I found the issue. In the [new v4 docs](https://v4.styledictionary.com/info/tokens/) it says either token format works. When changing `value`to `$value` though, the tokens actually compiles.

**Comment by jorenbroekema** [comment]
Ah right yes if you're using $type let's say, Style Dictionary assumes the rest of the tokens will be written in DTCG format, it's not possible to mix the DTCG format with the old SD JSON format 👍🏻 so it makes sense now why it didn't give you any output, because it didn't consider your tokens to be tokens

**Comment by Gristof** [comment]
Makes sense. I remember wanting to stick to the DTCG standard, but $value just wouldn't compile in v3. I probably should have removed all the $ instead of mix and match. Thanks for the explanation.

**Comment by jorenbroekema** [comment]
You can now switch to DTCG fully just FYI: https://v4.styledictionary.com/info/dtcg/
